Microsoft.Toolkit.Uwp.UI.Controls.Input/RichSuggestBox/RichSuggestBox.cs

```diff
@@ -0,0 +1,943 @@
+            {
+                return;
+            }
+
+            TokenSelected.Invoke(this, new RichSuggestTokenSelectedEventArgs
+            {
+                Token = token,
+                Range = selection.GetClone()
+            });
+        }
+
+        private void InvokeTokenHovering(PointerPoint pointer)
+        {
+            var pointerPosition = TransformToVisual(_richEditBox).TransformPoint(pointer.Position);
+            var padding = _richEditBox.Padding;
+            pointerPosition.X += HorizontalOffset - padding.Left;
+            pointerPosition.Y += VerticalOffset - padding.Top;
+            var range = TextDocument.GetRangeFromPoint(pointerPosition, PointOptions.ClientCoordinates);
+            var linkRange = range.GetClone();
+            range.Expand(TextRangeUnit.Character);
+            range.GetRect(PointOptions.None, out var hitTestRect, out _);
+            hitTestRect.X -= hitTestRect.Width;
+            hitTestRect.Width *= 2;
+            if (hitTestRect.Contains(pointerPosition) && linkRange.Expand(TextRangeUnit.Link) > 0 &&
+                TryGetTokenFromRange(linkRange, out var token))
+            {
+                this.TokenHovering.Invoke(this, new RichSuggestTokenHoveringEventArgs
+                {
+                    Token = token,
+                    Range = linkRange,
+                    CurrentPoint = pointer
+                });
+            }
+        }
+
+        private async Task RequestSuggestionsAsync(ITextRange range = null)
+        {
+            string prefix;
+            string query;
+            var queryFound = range == null
+                ? TryExtractQueryFromSelection(out prefix, out query, out range)
+                : TryExtractQueryFromRange(range, out prefix, out query);
+
+            if (queryFound && prefix == _currentPrefix && query == _currentQuery &&
+                range.EndPosition == _currentRange?.EndPosition && _suggestionPopup.IsOpen)
+            {
+                return;
+            }
+
+            CancelIfNotDisposed(this._suggestionRequestedCancellationSource);
+            this._suggestionRequestedCancellationSource = null;
+
+            if (queryFound)
+            {
+                _currentPrefix = prefix;
+                _currentQuery = query;
+                _currentRange = range;
+
+                if (SuggestionsRequested != null)
+                {
+                    using var tokenSource = new CancellationTokenSource();
+                    _suggestionRequestedCancellationSource = tokenSource;
+                    var cancellationToken = tokenSource.Token;
+                    var eventArgs = new SuggestionsRequestedEventArgs { Query = query, Prefix = prefix };
+                    try
+                    {
+                        await SuggestionsRequested.InvokeAsync(this, eventArgs, cancellationToken);
```

Comment:
What's the cancellation source for here? Where is this being monitored?

Generally you use the `Cancel` flag on an argument if you want the developer to be able to interrupt the default behavior. Like for the TokenizingTextBox it can be used to override our behavior and not add a token as we check if they developer didn't want to tokenize an item:

https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/5171b99c95323a704922a6bdc70a1a5d5eb650cf/Microsoft.Toolkit.Uwp.UI.Controls.Input/TokenizingTextBox/TokenizingTextBox.cs#L453-L459

Or similarly prevent one from being removed:

https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/5171b99c95323a704922a6bdc70a1a5d5eb650cf/Microsoft.Toolkit.Uwp.UI.Controls.Input/TokenizingTextBox/TokenizingTextBox.cs#L532-L538

I think the idea here is that if the dev is looking up remote resources, you want to know if they decide to bailout? But since you don't do anything else after this event, does it matter? You can still use the cancellation token with a regular `DeferredEventArgs` object. You should only use the `DeferredCancelEventArgs` if the dev should indicate to your control to not do the default behavior.

Reply:
I see, I think got confused `DeferredEventArgs` with `DeferredCancelEventArgs`. `SuggestionsRequestedEventArgs` should inherit `DeferredEventArgs` here.